src/components/NavBar/NavBar.js

```diff
@@ -46,14 +48,30 @@ const useStyles = makeStyles((theme) => ({
   }
 }));
 
+// selector named functions for lint rules makes it easier to re-use if needed.
+const NavBarSelector = (state) => state.navBar;
+
 export default function NavBar(props) {
-  // current active tag
-  // TODO: needs persist in redux
-  const [currentTab, setCurrentTab] = React.useState('Home');
+  // redux store for the active tab
+  const dispatch = useDispatch();
+  const navBar = useSelector(NavBarSelector);
+
+  // set active tab for page refreshes and copy past of url
+  useEffect(() => {
+    // get patch to check if path is different from state.
+    // this will happen when a user copy and pastes or uses a shareurl
+    // the tool will need to change the the tab accordingly
+    const locationPath = window.location.pathname.replace('/', '');
+
+    if (!navBar || navBar !== locationPath) {
+      dispatch(changeActiveTab(locationPath));
+    }
+  }, [dispatch, navBar]);
```

Comment:
why is useEffect state dependent on the dispatch hook?

Reply:
following the lint rule =`React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array `

Reply:
fixed in last commit

Reply:
will keep but we can keep this open for discussion